packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -1403,5 +1687,14 @@ private static bool TypeRequiresNullCheckInSerialization(CSharpType type)
 
             return false;
         }
+
+        private static bool IsAdditionalProperty(PropertyProvider property)
+        {
+            // additional properties should have backing fields and its' type should be a dictionary
+            if (property.BackingField == null || !property.Type.IsDictionary)
+                return false;
+            var span = property.Name.AsSpan();
+            return span.StartsWith("Additional", StringComparison.Ordinal) && span.EndsWith("Properties", StringComparison.Ordinal);
```

Comment:
Yeah I think our properties would get a number appended. I considered adding an API that would just tell us if a property is an additional property, though I think it would need to be public since we need to check this in the SCM plugin? Any concerns with that?

Reply:
I think we can go with the public property to unblock this. Maybe call it `IsAdditionalProperties`?

Reply:
sure.